### PR TITLE
[Filebeat] Add Proxy config to httpjson v2

### LINKED ISSUE #24615: Connect to Okta module of Filebeat via proxy
As I want to fetch logs from Okta, I am using filebeat 7.11 to do it. In the modules.d directory of filebeat, I can see configurations for Okta available.

I have entered the URL & credentials mentioned there. But now the problem is the URL is accessible via a proxy from my environment.
Where do I configure the proxy setting like
Proxy URL, Proxy Port, Proxy username, Proxy password, etc

Now had I used logstash http_poller input to connect to Okta API, below would have been my configuration,
```
urls => {
                        test2 => {
                                method => get
                                url => "https://okta/api/v1/logs"
                                headers => {
                                        Accept => "application/json"
                                        Authorization => "KEY*****************"
                                }
                        }
                }
                proxy => { host => "1.2.3.4" port => 4444 scheme => "http"}
                request_timeout => 60
                schedule => { cron => "0 * * * *"}
                codec => "json"
```
But I dont see any such options while configuring Okta in filebeat.

Any input how to achieve this would be really helpful.

Thanks,
@elastic-data-integration

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

Adds a `proxy_url` config option to the httpjson v2 input.  Potentially resolves 24615.

## Why is it important?

No way to specify a proxy to use for the httpjson v2 input.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

Resolves #24615

### CODE REVIEW

File: x-pack/filebeat/input/httpjson/internal/v2/config_request.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ type requestConfig struct {
 	RedirectMaxRedirects   int               `config:"redirect.max_redirects"`
 	RateLimit              *rateLimitConfig  `config:"rate_limit"`
 	Transforms             transformsConfig  `config:"transforms"`
+	ProxyURL               string            `config:"proxy_url"`
```
Comment (marc-gr) [inline x-pack/filebeat/input/httpjson/internal/v2/config_request.go:91]:
```suggestion
	ProxyURL               *urlConfig            `config:"proxy_url"`
```

If you change to the `urlConfig` type, if will handle the validation and parsing of the url automatically.

File: x-pack/filebeat/input/httpjson/internal/v2/config_request.go [hunk 2]
```diff
@@ -97,6 +98,13 @@ func (c requestConfig) getTimeout() time.Duration {
 	return *c.Timeout
 }
 
+func (c requestConfig) getProxyUrl() string {
```
Comment (marc-gr) [inline x-pack/filebeat/input/httpjson/internal/v2/config_request.go:101]: I think we do not need this function, just checking the value of the field should be enough I think.

File: x-pack/filebeat/module/okta/system/config/input.yml [hunk 3]
```diff
@@ -12,6 +12,11 @@ request.ssl: {{ .ssl | tojson }}
 request.timeout: {{ .http_client_timeout }}
 {{ end }}
 
+
+{{ if .ssl }}
```
Comment (marc-gr) [inline x-pack/filebeat/module/okta/system/config/input.yml:16]:
```suggestion
{{ if .proxy_url }}
```

**Review by marc-gr — CHANGES_REQUESTED** [review]
Thanks for the PR! On top of the suggested changes, an entry in the `CHANGELOG.next.asciidoc` is required, under the Bugfixes -> Filebeat section.

Let us know if you need any help to move forward.

File: x-pack/filebeat/module/okta/system/config/input.yml (hunk 3 above)
Comment (legoguy1000) [reply x-pack/filebeat/module/okta/system/config/input.yml]: done

**Comment by legoguy1000** [comment]
Should I add the `proxy_url` config option to all the other modules using the httpjsn v2 input?? I didn't want to right away until i got a thumbs up on the Golang code.

**Review by andrewkroh — APPROVED** [review]
Awesome, LGTM. Thanks!

~@marc-gr Can you check this too please.~

**Comment by andrewkroh** [comment]
jenkins, test this

File: CHANGELOG.next.asciidoc [hunk 4]
```diff
@@ -374,6 +374,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Fix `google_workspace` pagination. {pull}24668[24668]
 - Fix Cisco ASA parser for message 302022. {issue}24405[24405] {pull}24697[24697]
 - Fix date parsing in GSuite/login fileset. {issue}24694[24694]
+- Add `proxy_url` config for httpjson vs input. {issue}24615[24615] {pull}24662[24662]
```
Comment (marc-gr) [inline CHANGELOG.next.asciidoc:377]: This is under the Bugfixes section, could you please move it under the `Added` one?

**Review by marc-gr — CHANGES_REQUESTED** [review]
Thanks for addressing the changes! It LGTM except for the changelog line, LMK if you need any help with that 👍

**Comment by andrewkroh** [comment]
run tests

**Review by marc-gr — APPROVED** [review]

**Comment by andrewkroh** [comment]
jenkins, run tests

**Comment by andrewkroh** [comment]
run tests

**Pull request merged**